src/Materials/materialHelper.ts

```diff
@@ -577,7 +577,8 @@ export class MaterialHelper {
      * @param defines The current Defines of the effect
      */
     public static PrepareAttributesForInstances(attribs: string[], defines: any): void {
-        if (defines["INSTANCES"]) {
+        // Check if we are handling a MaterialDefine or an Array containing the #define
+        if (defines["INSTANCES"] || (defines instanceof Array && defines.indexOf("#define INSTANCES") > -1)) {

```

Comment:
We have to be cautious here
this call is executed on every frame for every mesh so we cannot afford having a indexOf because it is a slow call

Reply:
Thanks for the review! 

Are you thinking about just removing it and relying solely on the type check in case it is not a MaterialDefines, or did you have something else in mind?

I could not find a way of performing the check much faster aside from a reverse looping since we can assume the define got added somewhere at the end, but even that did not seem too great performance wise against indexOf depending on the browser used.

Reply:
I was thinking about removing it completely